lib/hooks/useMenu.tsx

```diff
@@ -46,14 +46,16 @@ export const useMenu = <T, V>({
 
     const flatItems: T[] = Object.values(grouped).flat();
 
-    return Object.entries(grouped).map(([key, groupItems]) => ({
-      items: groupItems.map((item) => ({
-        menuIndex: flatItems.indexOf(item),
-        active: activeIndex === flatItems.indexOf(item),
-        props: item,
-      })),
-      props: groups[key] || {},
-    }));
+    return Object.entries(grouped)
```

Comment:
At first glance:

Imho this is not a good solution to have the sorting in this component. It might work we sorting dates desc (if this is the case), but it would make sorting by months within a year difficult. I think useMenu hook should follow the natural order of the key map here. 

Could we discuss this?